File: wlmedia_hm/libwlmedia/src/main/ets/com/ywl5320/wlmedia/bean/WlTrackInfoBean.ts
import { WlTrackType } from "../enums/WlTrackType";

/**
 * author : ywl5320
 * e-mail : ywl5320@163.com
 * desc   : wlmedia
 * date   : 2025/1/10
 */
export class WlTrackInfoBean {
  trackType: WlTrackType;
  trackIndex: number;
  programIndex: number;
  duration: number;
  startTime: number;
  bitrate: number;
  variantBitrate: number;
  language: string;
  videoWidth: number;
  videoHeight: number;
  videoScaleWidth: number;
  videoScaleHeight: number;
  videoRotate: number;
  videoFrameRate: number;
  audioSampleRate: number;
  audioChannelNum: number;

  constructor() {
  }

  public getString(): string {
    return `
      Track Info:
      - Track Type: ${this.trackType}
      - Track Index: ${this.trackIndex}
      - Program Index: ${this.programIndex}
      - Duration: ${this.duration}
      - Start Time: ${this.startTime}
      - Bitrate: ${this.bitrate}
      - Variant Bitrate: ${this.variantBitrate}
      - Language: ${this.language}
      - Video Width: ${this.videoWidth}
      - Video Height: ${this.videoHeight}
      - Video Scale Width: ${this.videoScaleWidth}
      - Video Scale Height: ${this.videoScaleHeight}
      - Video Rotate: ${this.videoRotate}
      - Video Frame Rate: ${this.videoFrameRate}
      - Audio Sample Rate: ${this.audioSampleRate}
      - Audio Channel Number: ${this.audioChannelNum}
    `.trim();
  }

  toString(): string {
    return `
      Track Info:
      - Track Type: ${this.trackType}
      - Track Index: ${this.trackIndex}
      - Program Index: ${this.programIndex}
      - Duration: ${this.duration}
      - Start Time: ${this.startTime}
      - Bitrate: ${this.bitrate}
      - Variant Bitrate: ${this.variantBitrate}
      - Language: ${this.language}
      - Video Width: ${this.videoWidth}
      - Video Height: ${this.videoHeight}
      - Video Scale Width: ${this.videoScaleWidth}
      - Video Scale Height: ${this.videoScaleHeight}
      - Video Rotate: ${this.videoRotate}
      - Video Frame Rate: ${this.videoFrameRate}
      - Audio Sample Rate: ${this.audioSampleRate}
      - Audio Channel Number: ${this.audioChannelNum}
    `.trim();
  }
}

File: wlmedia_android/wlmedia/src/main/java/com/ywl5320/wlmedia/bean/WlTrackInfoBean.java
package com.ywl5320.wlmedia.bean;

import com.ywl5320.wlmedia.enums.WlTrackType;

/**
 * author : ywl5320
 * e-mail : ywl5320@163.com
 * desc   : wlmedia
 * date   : 2024/8/11
 */
public class WlTrackInfoBean {
    private int trackType;
    private int trackIndex;
    private int programIndex;
    private double duration;
    private double startTime;
    private long bitrate;
    private long variantBitrate;
    private String language;
    private String codecName;
    private int videoWidth;
    private int videoHeight;
    private int videoScaleWidth;
    private int videoScaleHeight;
    private double videoRotate;
    private double videoFrameRate;
    private int audioSampleRate;
    private int audioChannelNum;

    public WlTrackInfoBean() {
    }

    public WlTrackType getTrackType() {
        return WlTrackType.find(trackType);
    }

    public int getTrackIndex() {
        return trackIndex;
    }

    public int getProgramIndex() {
        return programIndex;
    }

    public double getDuration() {
        return duration;
    }

    public double getStartTime() {
        return startTime;
    }

    public long getBitrate() {
        return bitrate;
    }

    public long getVariantBitrate() {
        return variantBitrate;
    }

    public String getLanguage() {
        return language;
    }

    public String getCodecName() {
        return codecName;
    }

    public int getVideoWidth() {
        return videoWidth;
    }

    public int getVideoHeight() {
        return videoHeight;
    }

    public int getVideoScaleWidth() {
        return videoScaleWidth;
    }

    public int getVideoScaleHeight() {
        return videoScaleHeight;
    }

    public double getVideoRotate() {
        return videoRotate;
    }

    public double getVideoFrameRate() {
        return videoFrameRate;
    }

    public int getAudioSampleRate() {
        return audioSampleRate;
    }

    public int getAudioChannelNum() {
        return audioChannelNum;
    }

    @Override
    public String toString() {
        return "WlTrackInfoBean{" +
                "trackType=" + trackType +
                ", trackIndex=" + trackIndex +
                ", programIndex=" + programIndex +
                ", duration=" + duration +
                ", startTime=" + startTime +
                ", bitrate=" + bitrate +
                ", variantBitrate=" + variantBitrate +
                ", language='" + language + '\'' +
                ", codecName='" + codecName + '\'' +
                ", videoWidth=" + videoWidth +
                ", videoHeight=" + videoHeight +
                ", videoScaleWidth=" + videoScaleWidth +
                ", videoScaleHeight=" + videoScaleHeight +
                ", videoRotate=" + videoRotate +
                ", videoFrameRate=" + videoFrameRate +
                ", audioSampleRate=" + audioSampleRate +
                ", audioChannelNum=" + audioChannelNum +
                '}';
    }
}
